source/pins.cmake

```diff
@@ -0,0 +1,18 @@
+set(pinScript  "${CMAKE_CURRENT_LIST_DIR}/pins.py")
+set(pinFileDir "${CMAKE_CURRENT_LIST_DIR}/../generated")
```

Comment:
Not a real issue, but I'd put `generated/` under `build/`, to make it more clear that the directory contains automatically generated files that are not supposed to be edited. We already have `yotta_config.h` under `build/`.